src/toil/worker.py

```diff
@@ -539,18 +490,28 @@ def badWorker():
 def main(argv=None):
     if argv is None:
         argv = sys.argv
-        
+
     # Parse input args
     jobName = argv[1]
     jobStoreLocator = argv[2]
     jobStoreID = argv[3]
-    
+
     ##########################################
     #Load the jobStore/config file
     ##########################################
-    
+
+    # Try to monkey-patch boto early so that credentials are cached.
```

Comment:
why did you move this to the end of the file?

Reply:
I should have been clearer about this change. IMO it's best if this monkey-patch is run *before* any boto calls are made, so that we're using 1 single monkey-patched version of the boto code the entire time. Before this change boto was getting monkey-patched only after `Toil.resumeJobStore` was already called, which I believe contacts the jobStore. Obviously the code worked just fine before, so it doesn't actually matter in practice, but this seems a tiny bit cleaner to me.

(I could have put it at the top of the file as well, but that would monkey-patch any random process that [for some reason] imported stuff from the worker file.)